LibreNMS/OS/ArubaInstant.php

```diff
@@ -0,0 +1,224 @@
+     */
+    public function discoverWirelessPower()
+    {
+        // instant
+        return $this->discoverInstantRadio('power', 'aiRadioTransmitPower', "%s Radio %s: Tx Power");
+    }
+
+    /**
+     * Discover wireless utilization.  This is in %. Type is utilization.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessUtilization()
+    {
+        // instant
+        return $this->discoverInstantRadio('utilization', 'aiRadioUtilization64');
+    }
+
+    /**
+     * Aruba Instant Radio Discovery
+     *
+     * @return array Sensors
+     */
+    private function discoverInstantRadio($type, $mib, $desc = '%s Radio %s')
+    {
+        $ai_mib = 'AI-AP-MIB';
+        $ai_sg_data = $this->getCacheTable('aiStateGroup', $ai_mib);
+
+        $sensors = [];
+        foreach ($ai_sg_data as $ai_ap => $ai_ap_oid) {
+            if (isset($ai_ap_oid[$mib])) {
+                foreach ($ai_ap_oid[$mib] as $ai_ap_radio => $value) {
+                    if ($type == 'frequency') {
+                        $value = WirelessSensor::channelToFrequency($this->decodeChannel($value));
+                    }
+                    $combined_oid = sprintf('%s::%s.%s.%s', $ai_mib, $mib, $this->encodeMacToOid($ai_ap), $ai_ap_radio);
+                    $oid = snmp_translate($combined_oid, 'ALL', 'arubaos', '-On', null);
+                    $description = sprintf($desc, $ai_sg_data[$ai_ap]['aiAPSerialNum'], $ai_ap_radio);
+                    $index = sprintf('%s.%s', $this->encodeMacToHex($ai_ap), $ai_ap_radio);
+                    $sensors[] = new WirelessSensor($type, $this->getDeviceId(), $oid, 'aruba-instant', $index, $description, $value);
+                } // end foreach
+            } // end if
+        } // end foreach
+        return $sensors;
+    }
+
+    protected function decodeChannel($channel)
+    {
+        return $channel & 255; // mask off the channel width information
+    }
+
+    /**
+     * Convert Hex MAC to oid MAC
+     * 00:12:34:AB:CD:EF becomes 0.18.52.171.205.239
+     *
+     * @param string $mac hexadecimal MAC address with or without standard delimiters
+     * @return string oid representation of a MAC address
+     */
+    protected function encodeMacToOid($mac)
+    {
+        $oid = '';
+        $macparts = explode(':', $mac);
+        foreach ($macparts as $part) {
+            $oid .= hexdec($part).'.';
+        }
+        return rtrim($oid, '.');
+    }
+
+    /**
+     * Convert Hex MAC to Hex String without delimiters
+     * 00:12:34:AB:CD:EF becomes 001234ABCDEF
+     * 0:12:34:AB:CD:EF  becomes 001234ABCDEF
+     * 0:2:4:B:D:F       becomes 0002040B0D0F
+     *
+     * @param string $mac hexadecimal MAC address with or without standard delimiters
+     * @return string oid representation of a MAC address
+     */
+    protected function encodeMacToHex($mac)
+    {
+        $hex = '';
```

Comment:
This seems to be done a lot like this:
`implode(array_map('zeropad', explode(':', $raw_mac)))`